### PR TITLE
IMPRO-1485: style fixing

### PR DESCRIPTION
**Pull request by neilCrosswaite**
Addressed some style changes or more consistent code.

Areas changed:
- unused imports
- vars outside of loops 
  - using next()
- no else after raise,return,continue
- using an if statement to assign True or False
  - `result = var == 1` will work
- `!=` instead of `not... ==.. `
- removing the use of `keys()`
- disbled `cube.data` for no lazy loading
- set comprehension rather than `set()` with a list comprehension in

Testing:
 - [x] Ran tests and they passed OK

### CODE REVIEW

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 10
           

Complexity decreasing per file
==============================
+ improver/cli/spot_extract.py  -1
+ improver/generate_ancillaries/generate_topographic_zone_weights.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4838980&bid=16068812)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 22
           

Complexity decreasing per file
==============================
+ improver/generate_ancillaries/generate_topographic_zone_weights.py  -1
+ improver/cli/spot_extract.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4838980&bid=16068812)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 30
           

Complexity decreasing per file
==============================
+ improver/generate_ancillaries/generate_topographic_zone_weights.py  -1
+ improver_tests/utilities/test_cube_checker.py  -1
+ improver/cli/spot_extract.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4838980&bid=16068812)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 30
           

Complexity decreasing per file
==============================
+ improver_tests/utilities/test_cube_checker.py  -1
+ improver/cli/spot_extract.py  -1
+ improver/generate_ancillaries/generate_topographic_zone_weights.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4838980&bid=16068812)

File: improver/utilities/ancillary_creation.py [hunk 1]
```diff
@@ -28,6 +28,7 @@
 # CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
 # ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
+"""A module for creating ancillarys"""
```
Comment (MoseleyS) [inline improver/utilities/ancillary_creation.py]:
```suggestion
"""A module for creating ancillary data"""
```

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 30
           

Complexity decreasing per file
==============================
+ improver/generate_ancillaries/generate_topographic_zone_weights.py  -1
+ improver/cli/spot_extract.py  -1
+ improver_tests/utilities/test_cube_checker.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4838980&bid=16068812)

**Pull request merged**